registrations/tasks.py

```diff
@@ -420,22 +424,9 @@ def get_dob(self, mom_dob):
         else:
             return None
 
-    def get_authority_from_source(self, source):
-        """
-        NOTE:   this is a convenience method to map the new "source"
-                back to ndoh-control's "authority" fields to maintain
-                backwards compatibility with existing APIs
-        """
-        return {
-            'CLINIC USSD app': 'clinic',
-            'PUBLIC USSD app': 'personal',
-            'CHW USSD app': 'chw',
-            'NURSE USSD App': 'nurseconnect',
-        }.get(source.name, 'unknown')  # See note at get_subscription_type
-
     def build_jembi_json(self, registration):
         """ Compile json to be sent to Jembi. """
-        authority = self.get_authority_from_source(registration.source)
+        authority = registration.source.authority
```

Comment:
The registration source model does have an authority field but it contains data like "hw_limited" or "hw_full" or "patient". It doesn't directly map to the system (line / app) as different systems can have the same authority level.